### DISCUSSION TITLE
Webpack 4 treeshakes firebase 8 auth import off

### DISCUSSION THREAD

**Discussion by jancama2** [webpack/webpack#15001] [Q&A]
I'm using Webpack with a really basic configuration with optiomization turned on in production and with TerserPlugin as minifier. When i set `"sideEffects": false`, Webpack treeshakes imports of firebase 8 services from the app.

```
import firebase from 'firebase/app'

import 'firebase/auth'; // this one is missing in production
import 'firebase/firestore'; // this one is missing in production too

firebase.initializeApp(config);

export {
   firebase,
}

```

I thought Webpack handles `node_modules` differently and doesn't treeshake them but looks like I'm wrong.
I tried to put the modules to sideeffects `"sideEffects": ["firebase/auth", "firebase/firestore"]` but it didn't help.

Could anyone point me where the problem is? 🙏

**Comment by alexander-akait** [comment]
Can you try webpack v5? Because a lot of bugs were fixed, we need verify it, anyway if you provide minimum example it will be easier to show

**Reply by jancama2** [reply]
Will try Webpack 5 👌  thanks 🙂
